module/VuFind/src/VuFind/Navigation/AccountMenuFactory.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * Account menu factory
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Navigation
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+
+namespace VuFind\Navigation;
+
+use Laminas\ServiceManager\Exception\ServiceNotCreatedException;
+use Laminas\ServiceManager\Exception\ServiceNotFoundException;
+use Psr\Container\ContainerExceptionInterface as ContainerException;
+use Psr\Container\ContainerInterface;
+
+/**
+ * Account menu factory
+ *
+ * @category VuFind
+ * @package  Navigation
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class AccountMenuFactory extends AbstractMenuFactory
+{
+    /**
+     * Create an object
+     *
+     * @param ContainerInterface $container     Service manager
+     * @param string             $requestedName Service being created
+     * @param null|array         $options       Extra options (optional)
+     *
+     * @return object
+     *
+     * @throws ServiceNotFoundException if unable to resolve the service.
+     * @throws ServiceNotCreatedException if an exception is raised when
+     * creating a service.
+     * @throws ContainerException&\Throwable if any other error occurs
+     */
+    public function __invoke(
+        ContainerInterface $container,
+        $requestedName,
+        array $options = null
+    ) {
+        // Only load the connector if we need to show
+        $config = $container->get(\VuFind\Config\PluginManager::class)->get(
+            'Overdrive'
+        );
+        $connector = null;
+        $showMyContent = $config->Overdrive->showMyContent;
+        $showAdmin = $config->Overdrive->showOverdriveAdminMenu;
```

Comment:
Probably worth null coalescing some defaults here... and maybe even calling `toArray()` above and using arrays, to help with our necessary migration away from Laminas\Config\Config.

Reply:
Also, why does the showOverdriveAdminMenu come into play here at all? Is this refactored from somewhere else? My understanding is that this setting should only impact the admin menu and not the account menu, but maybe I'm misunderstanding something.

Reply:
You are right, the code was copied from `OverdriveFactory` and `showOverdriveAdminMenu` is not relevant here.